src/NClient/NClient.Standalone/ClientProxy/Validation/PipelineCanceller.cs

```diff
@@ -0,0 +1,30 @@
+﻿using NClient.Providers.Transport.Common;
+using System.Threading;
+
+namespace NClient.Standalone.ClientProxy.Validation
+{
+    internal class PipelineCanceller : IPipelineCanceller
+    {
+        internal CancellationTokenSource _canceller;
```

Comment:
`CancellationTokenSource` is disposable class, so you have to dispose it after use

Reply:
Good spot! I added Dispose() call when renewing the `cancellationTokenSource`, I'm not clear on if dispose should be called elsewhere because the lifecycle of the objects that use The `PipelineCanceller` aren't under our control. ie the clients are returned out of the framework to the dev's calling .Build(). Have you got any idea's about other places where dispose would be appropriate (your knowledge of the frameworks lifecycle is better than mine 😄)